pennylane/templates/subroutines/ctrlsequence.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the CtrlSequence template.
+"""
+from copy import copy
+import numpy as np
+import pennylane as qml
+
+from pennylane.operation import AnyWires
+from pennylane.wires import Wires
+from pennylane.ops.op_math.symbolicop import SymbolicOp
+
+
+class CtrlSequence(SymbolicOp):
```

Comment:
I didn't know `SymbolicOp` but I would say it makes sense

Reply:
What makes this naturally a symbolic op?

Reply:
That it's an Operator that is initialized using another operator as its base. 

The `SymbolicOp` class inherits from `Operator`, and adds methods to make sure the base is copied when copying the operator, to inherit relevant properties from the base, and to avoid queueing the base when initializing, even though it was technically called inside the qnode.

The `Controlled` class is also a `SymbolicOp`, for example.

Reply:
But possibly we want to inherit from `(SymbolicOp, Operation)` instead

Reply:
Thanks, this makes a lot of sense! As an aside then, I'm not sure about the name `SymbolicOp` :thinking:

Reply:
It's explicitly developer-facing, so I guess we can just change it if we want to